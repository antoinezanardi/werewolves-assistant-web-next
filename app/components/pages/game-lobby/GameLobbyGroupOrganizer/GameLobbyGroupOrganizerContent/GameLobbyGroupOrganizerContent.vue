<template>
  <div
    id="game-lobby-group-organizer-content"
    class="flex flex-col gap-1"
  >
    <GameLobbyGroupOrganizerContentHeader/>

    <PrimeVueDivider class="!my-2"/>

    <GameLobbyGroupOrganizerGroupNames/>

    <GameLobbyGroupOrganizerPickList class="h-3/5"/>

    <GameLobbyGroupOrganizerDisclaimer class="flex grow items-center justify-center"/>
  </div>
</template>

<script setup lang="ts">
import GameLobbyGroupOrganizerContentHeader from "~/components/pages/game-lobby/GameLobbyGroupOrganizer/GameLobbyGroupOrganizerContent/GameLobbyGroupOrganizerContentHeader/GameLobbyGroupOrganizerContentHeader.vue";
import GameLobbyGroupOrganizerDisclaimer from "~/components/pages/game-lobby/GameLobbyGroupOrganizer/GameLobbyGroupOrganizerContent/GameLobbyGroupOrganizerDisclaimer/GameLobbyGroupOrganizerDisclaimer.vue";
import GameLobbyGroupOrganizerGroupNames from "~/components/pages/game-lobby/GameLobbyGroupOrganizer/GameLobbyGroupOrganizerContent/GameLobbyGroupOrganizerGroupNames/GameLobbyGroupOrganizerGroupNames.vue";
import GameLobbyGroupOrganizerPickList from "~/components/pages/game-lobby/GameLobbyGroupOrganizer/GameLobbyGroupOrganizerContent/GameLobbyGroupOrganizerPickList/GameLobbyGroupOrganizerPickList.vue";
</script>