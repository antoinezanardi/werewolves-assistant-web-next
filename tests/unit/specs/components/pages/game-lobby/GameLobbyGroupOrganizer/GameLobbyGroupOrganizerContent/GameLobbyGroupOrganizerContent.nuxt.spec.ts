import type { mount } from "@vue/test-utils";

import { mountSuspendedComponent } from "@tests/unit/utils/helpers/mount.helpers";
import type { ComponentMountingOptions } from "@vue/test-utils/dist/mount";
import GameLobbyGroupOrganizerContent from "~/components/pages/game-lobby/GameLobbyGroupOrganizer/GameLobbyGroupOrganizerContent/GameLobbyGroupOrganizerContent.vue";

describe("Game Lobby Group Organizer Content Component", () => {
  let wrapper: ReturnType<typeof mount<typeof GameLobbyGroupOrganizerContent>>;

  async function mountGameLobbyGroupOrganizerContentComponent(options: ComponentMountingOptions<typeof GameLobbyGroupOrganizerContent> = {}):
  Promise<ReturnType<typeof mount<typeof GameLobbyGroupOrganizerContent>>> {
    return mountSuspendedComponent(GameLobbyGroupOrganizerContent, { ...options });
  }

  beforeEach(async() => {
    wrapper = await mountGameLobbyGroupOrganizerContentComponent();
  });

  it("should match snapshot when rendered.", () => {
    expect(wrapper).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });
});