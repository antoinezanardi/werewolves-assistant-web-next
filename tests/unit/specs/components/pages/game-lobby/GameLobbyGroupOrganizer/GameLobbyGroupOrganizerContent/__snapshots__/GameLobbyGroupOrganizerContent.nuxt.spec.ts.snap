// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Game Lobby Group Organizer Content Component > should match snapshot when rendered. 1`] = `
"<div id="game-lobby-group-organizer-content" class="flex flex-col gap-1">
  <game-lobby-group-organizer-content-header-stub></game-lobby-group-organizer-content-header-stub>
  <divider-stub class="!my-2"></divider-stub>
  <game-lobby-group-organizer-group-names-stub></game-lobby-group-organizer-group-names-stub>
  <game-lobby-group-organizer-pick-list-stub class="h-3/5"></game-lobby-group-organizer-pick-list-stub>
  <game-lobby-group-organizer-disclaimer-stub class="flex grow items-center justify-center"></game-lobby-group-organizer-disclaimer-stub>
</div>"
`;
